internal/command/console.go

```diff
@@ -179,26 +179,38 @@ func (c *ConsoleCommand) Run(args []string) int {
 }
 
 func (c *ConsoleCommand) modePiped(session *repl.Session, ui cli.Ui) int {
-	var lastResult string
 	scanner := bufio.NewScanner(os.Stdin)
+
+	var consoleState consoleBracketState
+
 	for scanner.Scan() {
-		result, exit, diags := session.Handle(strings.TrimSpace(scanner.Text()))
-		if diags.HasErrors() {
-			// In piped mode we'll exit immediately on error.
-			c.showDiagnostics(diags)
-			return 1
-		}
-		if exit {
-			return 0
+		line := strings.TrimSpace(scanner.Text())
+
+		// we update the state with the new line, so if we have open
+		// brackets we know not to execute the command just yet
+		consoleState.UpdateState(line)
+
+		// we check if there is no escaped new line at the end, or any open brackets
+		// if we have neither, then we can execute
+		if !strings.HasSuffix(line, "\\") && consoleState.BracketsOpen() <= 0 && len(line) != 0 {
```

Comment:
I need to download and play with it, but I'm curious about the len(line) != 0:
```
> 5 + 4 \\

```

Wouldn't that get stuck?

Reply:
> 4 + 5
9
> 4 + 5 \\
 
╷
│ Error: Invalid character
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ This character is not used within the language.
╵

╷
│ Error: Extra characters after expression
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ An expression was successfully parsed, but extra characters were found after it.
╵

Reply:
wait... isn't `\\` just escaping the slash, not the new line. So the command we send to execute is `4 + 5` with a random slash after it.. which is invalid..